### PR TITLE
[SiPixel Payload Inspector] Quality Maps Improved

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The main goal of this PR is to improve the graphical appearance of plots produced with the pixel payload inspector, specifically for the `SiPixelQuality` flavour.
In doing so the class `Phase1PixelROCMaps` had to be changed in order to:
   * print the name of the tag and IOVs in the plot header (commit https://github.com/cms-sw/cmssw/commit/69821a20ff3ad43cc2190be44f7487c3df1efcf3);
   * support the case of diffing two payloads (commits 56dba64 and 85c9df339b45aea7b8296dda11b4c01b14db378f); 
   
All classes depending on that had to be modified accordingly (done in commits 2bb3909, https://github.com/cms-sw/cmssw/commit/ef689cd86748caea0384661aa8b7993763adde4c,  6e9c712)
I profit of this PR to do some additional clean-up:
   * update the scram architecture in all the test scripts (commit ea3a58c )
   * mark all the methods implemented in the Phase1PixelROCMaps header as `inline`d (see PR #33366) (commit https://github.com/cms-sw/cmssw/pull/34071/commits/f52b3946d61a683889e34d940510160ceda3e44e)
   * fix a mistake in one of the comments (commit b5ddb04db2d64355d94737c6cff23d05499f3f07 )

#### PR validation:

Unit tests of the package run successfully.
In addition run the following script:
```bash
#!/bin/sh

W_DIR=$(pwd)
mkdir -p $W_DIR/myplots

getPayloadData.py --plugin pluginSiPixelQuality_PayloadInspector --plot plot_SiPixelFullQualityMap --tag SiPixelQuality_phase1_2018_permanentlyBad --time_type Lumi --iovs '{"start_iov": "1", "end_iov": "1
"}' --db Prod --test ;
mv *.png $W_DIR/myplots/SiPixelQuality_phase1_2018_permanentlyBad.png

getPayloadData.py --plugin pluginSiPixelQuality_PayloadInspector --plot plot_SiPixelFullQualityMap --tag SiPixelQuality_phase1_2021_v2 --time_type Lumi --iovs '{"start_iov": "1", "end_iov": "1"}' --db Pro
d --test ;
mv *.png $W_DIR/myplots/SiPixelQuality_phase1_2021_v2.png

getPayloadData.py --plugin pluginSiPixelQuality_PayloadInspector --plot plot_SiPixelFullQualityMapCompareTwoTags --tag SiPixelQuality_phase1_2021_v2 --tagtwo SiPixelQuality_phase1_2018_permanentlyBad --ti
me_type Run --iovs '{"start_iov": "1", "end_iov": "1"}' --iovstwo '{"start_iov": "1", "end_iov": "1"}' --db Prod --test ;
mv *.png $W_DIR/myplots/SiPixelQuality_diff.png

getPayloadData.py --plugin pluginSiPixelQuality_PayloadInspector --plot plot_SiPixelBPixQualityMapCompareTwoTags --tag SiPixelQuality_phase1_2021_v2 --tagtwo SiPixelQuality_phase1_2018_permanentlyBad --ti
me_type Run --iovs '{"start_iov": "1", "end_iov": "1"}' --iovstwo '{"start_iov": "1", "end_iov": "1"}' --db Prod --test ;
mv *.png $W_DIR/myplots/SiPixelQuality_BPIX_diff.png

getPayloadData.py --plugin pluginSiPixelQuality_PayloadInspector --plot plot_SiPixelFPixQualityMapCompareTwoTags --tag SiPixelQuality_phase1_2021_v2 --tagtwo SiPixelQuality_phase1_2018_permanentlyBad --ti
me_type Run --iovs '{"start_iov": "1", "end_iov": "1"}' --iovstwo '{"start_iov": "1", "end_iov": "1"}' --db Prod --test ;
mv *.png $W_DIR/myplots/SiPixelQuality_FPIX_diff.png
```
and obtained [the following plots](http://musich.web.cern.ch/musich/display/myplots/).

I post one specific example here:
image

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Not a backport, no backport is needed.

CC: @tvami @tsusa

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: DQM/TrackerRemapper/src/Phase1PixelROCMaps.cc [hunk 1]
```diff
@@ -88,7 +88,7 @@ rocBins Phase1PixelROCMaps::maskedBarrelRocsToBins(DetCoordinates coord, std::bi
     if (myRocs.test(idx)) {
       //////////////////////////////////////////////////////////////////////////////////////
       //		                        |					  //
-      // In BPix Layer1 and module>0 in L2,3,4  |   In BPix Layer 2,3,4 module > 0	  //
+      // In BPix Layer1 and module<0 in L2,3,4  |   In BPix Layer 2,3,4 module > 0	  //
```
Comment (ferencek) [inline DQM/TrackerRemapper/src/Phase1PixelROCMaps.cc:91]: Wasn't the one on the left ok and instead the one on the right should be fixed?

File: DQM/TrackerRemapper/src/Phase1PixelROCMaps.cc [hunk 2]
```diff
@@ -88,7 +88,7 @@ rocBins Phase1PixelROCMaps::maskedBarrelRocsToBins(DetCoordinates coord, std::bi
...
       //                                        |					  //
       // ROCs are ordered in the following      |   ROCs are ordered in the following     //
       // fashion for unplipped modules   	|   fashion for unplipped modules         //
```
Comment (ferencek) [inline DQM/TrackerRemapper/src/Phase1PixelROCMaps.cc:94]: unflipped

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by jfernan2** [comment]
@mmusich  might this PR affect Online DQM plots somehow or it just modifies those generated by the  getPayloadData.py script?
Thanks

**Comment by mmusich** [comment]
@jfernan2 it does not affect Online DQM, only plots generated with the `getPayloadData.py` script (either privately or in the CondDB browser).

**Comment by jfernan2** [comment]
+1

**Comment by qliphy** [comment]
kindly ping @cms-sw/db-l2

**Comment by ggovi** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**